src/overlays/misc/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -516,166 +526,177 @@ void KaleidoScope_DrawQuestStatus(PlayState* play, GraphicsContext* gfxCtx) {
+                if (pauseCtx->mainState == PAUSE_MAIN_STATE_IDLE_CURSOR_ON_SONG) {
                     if (gOcarinaSongButtons[stepGreen].buttonsIndex[i] == OCARINA_BTN_A) {
-                        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, KALEIDO_COLOR_COLLECT_UNK_R, KALEIDO_COLOR_COLLECT_UNK_G,
-                                        KALEIDO_COLOR_COLLECT_UNK_B, 200);
+                        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, QUEST_OCARINA_BTN_A_COLOR_R, QUEST_OCARINA_BTN_A_COLOR_G,
+                                        QUEST_OCARINA_BTN_A_COLOR_B, 200);
                     } else {
                         gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 50, 200);
                     }
                 } else {
+                    // Gray out buttons during the player playing the song
                     gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 150, 150, 150, 150);
                 }
 
                 gDPSetEnvColor(POLY_OPA_DISP++, 10, 10, 10, 0);
 
                 gSPVertex(POLY_OPA_DISP++, &pauseCtx->questVtx[bufI], 4, 0);
 
-                gDPLoadTextureBlock(POLY_OPA_DISP++, D_8082A130[gOcarinaSongButtons[stepGreen].buttonsIndex[i]],
-                                    G_IM_FMT_IA, G_IM_SIZ_8b, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP,
-                                    G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+                gDPLoadTextureBlock(POLY_OPA_DISP++,
+                                    sOcarinaBtnTextures[gOcarinaSongButtons[stepGreen].buttonsIndex[i]], G_IM_FMT_IA,
+                                    G_IM_SIZ_8b, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP,
+                                    G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
 
                 gSP1Quadrangle(POLY_OPA_DISP++, 0, 2, 3, 1, 0);
             }
 
-            if (pauseCtx->mainState != PAUSE_MAIN_STATE_8) {
+            if (pauseCtx->mainState != PAUSE_MAIN_STATE_IDLE_CURSOR_ON_SONG) {
+                // Draw the buttons colored as the player plays the song
+
                 pauseCtx->ocarinaStaff = AudioOcarina_GetPlayingStaff();
 
                 if (pauseCtx->ocarinaStaff->pos != 0) {
-                    if (D_8082A11C == (pauseCtx->ocarinaStaff->pos - 1)) {
+                    if (sPlayedSongBtnsNum == (pauseCtx->ocarinaStaff->pos - 1)) {
                         if ((pauseCtx->ocarinaStaff->buttonIndex >= OCARINA_BTN_A) &&
                             (pauseCtx->ocarinaStaff->buttonIndex <= OCARINA_BTN_C_UP)) {
-                            D_8082A124[pauseCtx->ocarinaStaff->pos - 1] = pauseCtx->ocarinaStaff->buttonIndex;
-                            D_8082A124[pauseCtx->ocarinaStaff->pos] = 0xFF;
-                            D_8082A11C++;
+                            sPlayedSongBtns[pauseCtx->ocarinaStaff->pos - 1] = pauseCtx->ocarinaStaff->buttonIndex;
+                            sPlayedSongBtns[pauseCtx->ocarinaStaff->pos] = OCARINA_BTN_INVALID;
```

Comment:
I haven't looked too deeply into this, but are we sure `pauseCtx->ocarinaStaff->pos` can never be 8 here to indicate "one past the end"? The `pos - 1` stuff here made me suspicious, and glancing at the audio code it seems like pos can be 8 after `AudioOcarina_UpdatePlaybackStaff` but not sure.

Maybe sPlayedSongBtns/sPlayedSongBtnsAlpha should have size `SONG_MAX_LENGTH + 1`? That could get rid of the suspicious unused statics too.

Reply:
`pauseCtx->ocarinaStaff->pos` is in fact 8 here when the player inputs the last note of an 8-notes song like bolero, thanks